src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -0,0 +1,829 @@
+            size="small"
+            title="Copy"
+            data-testid="k-code-block-copy-button"
+            @click="copyToClipboard(props.code)"
+          >
+            <KIcon
+              icon="copy"
+              size="24"
+              title="Copy"
+              color="currentColor"
+            />
+
+            <span class="k-visually-hidden">Copy</span>
+          </KButton>
+
+          <template #content>
+            Code copied to clipboard
+          </template>
+        </KPop>
+      </KClipboardProvider>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, nextTick, onBeforeUnmount, onMounted, ref, watch } from 'vue'
+
+import KButton from '@/components/KButton/KButton.vue'
+import KClipboardProvider from '@/components/KClipboardProvider/index'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KInput from '@/components/KInput/KInput.vue'
+import KLabel from '@/components/KLabel/KLabel.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import { debounce } from '@/utils/debounce'
+
+export type CodeBlockEventData = {
+  preElement: HTMLElement
+  codeElement: HTMLElement
+  code: string
+  language: string
+  query: string
+  matchingLineNumbers: number[]
+}
+
+// Debounces the search handler which ensures that we don’t trigger several searches while the user is still typing.
+const debouncedHandleSearchInputValue = debounce(handleSearchInputValue, 150)
+
+const props = defineProps({
+  /**
+   * ID value used for form elements like the search input and its label.
+   */
+  id: {
+    type: String,
+    required: false,
+    default: 'code-block',
+  },
+
+  /**
+   * The code that will be rendered as a text node inside of a `<pre><code></code></pre>` fragment.
+   */
+  code: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * The syntax language of `props.code` (e.g. `'json'`).
+   */
+  language: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * Shows an actions bar with a search input and related action buttons.
+   */
+  isSearchable: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+
+  /**
+   * Allows controlling the processing state from outside the component. This allows a parent component to show the processing icon when it’s, for example, currently syntax highlighting the code.
+   */
+  isProcessing: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+
+  /**
+   * Used as the initial value of the search input. Can be used to initialize a code block with a query which was read from client storage.
+   */
+  query: {
+    type: String,
+    required: false,
+    default: '',
+  },
+
+  /**
+   * Controls whether to show a copy button. It copies the original code (i.e. the value of `props.code`) to the clipboard.
+   */
+  showCopyButton: {
+    type: Boolean,
+    required: false,
+    default: true,
+  },
+
+  /**
+   * Controls whether to show line numbers.
+   */
+  showLineNumbers: {
```

Comment:
Let's rename this prop to `lineNumbers` as we're trying to get away from prefixed props, e.g. `show*`